readme.md

```diff
@@ -2,25 +2,81 @@
 
 Rainbow bridge for shipping your files to any cloud storage service with the same function calls.
 
-![image](https://user-images.githubusercontent.com/59074379/226159115-1cfcb221-127f-4574-87ed-b74b4b2c4591.png)
+<img src="https://user-images.githubusercontent.com/59074379/226159115-1cfcb221-127f-4574-87ed-b74b4b2c4591.png" width="1000" />
+
+# Table of contents
+
+- [Bifrost](#bifrost)
+- [Problem Statement](#problem-statement)
+  - [Google Cloud Storage(GCS)](#google-cloud-storagegcs)
+  - [Pinata](#pinata)
+- [Installation](#installation)
+- [Usage](#usage)
+- [Contributing](#contributing)
+- [License](#license)
+- [Changelog](#changelog)
+- [Contributors](#contributors)
 
 # Problem Statement
+Many projects need to store files in the cloud, and different projects might use different cloud storage providers. Using different SDKs with different implementations for each provider can be tedious and time-consuming. Bifrost aims to simplify the process of working with multiple cloud storage providers by providing a consistent API for all of them. To better understand how Bifrost solves this problem, let's take a look at two separate code samples for GCS and Pinata using conventional means and how Bifrost eases the steps. 
 
-You might just want to use 3 different cloud storage providers in your project. This means you'll need 3 different SDKs with 3 different implementations. That's just too much learning curve.
 
-How about you ride with Thor on the Bifrost and easily transport your files to any cloud storage with the exact same function calls.
+## Google Cloud Storage(GCS)
```

Comment:
For this, I think the presentation here doesn't appear as in the pattern of the problem statement. Bifrost code even looks longer...haha. 

The idea is that we should create two sample codes in sequence... 

## Google Cloud Storage using GCS SDK
```go
// insert code sample sing GCS SDK (same one you already have)
```

## Pinata Cloud using Pinata API
```go
// insert code sample using Pinata API (same one you already have)
```

Then we add a line saying something like
> From the sample above, it's easy to see that using both GCS and Pinata in your Go project will require you to learn to use two different tools with separate implementation patterns. Now, this is where Bifrost comes in. Mount bridges to the providers you want and call the exact same function when you need to upload files using any of these bridges. Now let's rewrite the above the Bifrost way. 

```go
package main

import (
	"fmt"
	"os"
	"github.com/opensaucerer/bifrost"
) 

// mount a bridge to gcs
gcsBridge, _ := bifrost.NewRainbowBridge(&bifrost.BridgeConfig{
	DefaultBucket:   "bifrost",
	DefaultTimeout:  10,
	Provider:        bifrost.GoogleCloudStorage,
	CredentialsFile: "/path/to/service/account/json", // this is not required if you are using google's default credentials
	EnableDebug:     true,
	PublicRead:      true,
})
defer gcsBridge.Disconnect()
fmt.Printf("Connected to %s\n", bridge.Config().Provider)

// mount a bridge to Pinata
pinataBridge, _ := bifrost.NewRainbowBridge(&bifrost.BridgeConfig{
	Provider:    bifrost.PinataCloud,
	PinataJWT:   os.Getenv("PINATA_JWT"),
	EnableDebug: true,
	PublicRead:  true,
})
defer pinataBridge.Disconnect()
fmt.Printf("Connected to %s\n", bridge.Config().Provider)

// upload a file to gcs using the bridge
guf, _ := gcsBridge.UploadFile(bifrost.File{
	Path:     "../shared/image/aand.png",
	Filename: "a_and_ampersand.png",
	Options: map[string]interface{}{
		bifrost.OptMetadata: map[string]string{
			"originalname": "aand.png",
		},
	},
})
fmt.Printf("Uploaded to GCS file: %s to %s\n", guf.Name, guf.Preview)

// upload a file to Pinata using the bridge
puf, _ := bridge.UploadFile(bifrost.File{
	Path:     "../shared/image/aand.png",
	Filename: "pinata_aand.png",
	Options: map[string]interface{}{
		bifrost.OptPinata: map[string]interface{}{
			"cidVersion": 1,
		},
		bifrost.OptMetadata: map[string]string{
			"originalname": "aand.png",
		},
	},
})
fmt.Printf("Uploaded to Pinata file: %s to %s\n", puf.Name, puf.Preview)
```

Then, after the above, we can add another closing line saying something like
> From the above example, it's easy to see the speed, simplicity, and ease of use that comes with Bifrost. Now you know what it feels like to ride with Thor!

Reply:
I've resolved the requested changes and it's ready for review.